### PR TITLE
PCHR-3061: Task and Assignment admin styles

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
The PR(and https://github.com/compucorp/org.civicrm.shoreditch/pull/84, https://github.com/compucorp/civihr-employee-portal-theme/pull/315) fixes the T&A Admin styles.

## Problem 1
Select dates with no tasks assigned , See blank table without any notifications
Select dates with no existing key dates, See blank table without any notifications
Select dates with no documents assigned, See blank table with headers and no notifications

### Before
2018-02-05 at 1 53 pm

### After
2018-02-05 at 2 03 pm

Similarly text is shown in Documents and Key Dates

### Technical Details
1. No Result Text added to `dashboard/documents.html`, `dashboard/key-dates.html` and `dashboard/tasks.html`.

2. Also added margin top to `partials/_lists.scss` to provide enough space on of top no result text.

## Problem 2
Click 'Add workflow' button, See that Key Date field is highlighted with red as if it was wrong

### Before
2018-02-05 at 1 59 pm

### After
2018-02-05 at 2 03 pm

### Technical Details
In `modal/assignment.html`, added the `has-error` was being applied before the form is added. So added `assignmentForm.submitted` condition.

## Problem 3
In Tasks, Workflow Type dropdown on hover, the text is not visible.

### Before
2018-02-02 at 2 25 pm

### After
2018-02-05 at 2 04 pm

### Technical Details
The UI Select for the Admin theme should look like the one present in `/civicrm/contact/search?reset=1`. So made necessary changes.
1. There were few places on T&A where `theme="civihr-ui-select"` was applied to the `ui-select` dropdown. Added the same in all missing places.

2. Also added the following css to `_civihr-ui-select.scss`, to fix the smaller font size(11px) when ui-select is in focus.

```
.select2-search-choice {
  font-size: 13px;
}
```

it overrides the css from `sites/all/modules/civicrm/css/civicrm.css`
```
.crm-container div.select2-container {
  font-size: 11px;
}
```

3. Fixed the border color for the selected Multi Select Values `border-color: $brand-primary !important;`.

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss [hunk 1]
```diff
@@ -7,6 +7,14 @@
 
 // Customizations specific for T&A
 .civihr-ui-select {
+  * {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss]: is this really necessary? it's quite a wide net you are casting here

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss [hunk 2]
```diff
@@ -7,6 +7,14 @@
...
+    border-color: $brand-primary !important;
+  }
+
+  .select2-search-choice {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss]: why is this necessary? where did that 11px font size come from?

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_lists.scss [hunk 3]
```diff
@@ -3,6 +3,10 @@
   margin: 0;
   padding: 0;
 
+  > div {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_lists.scss]: what's the markup here? You don't have a class you can target?

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html [hunk 4]
```diff
@@ -134,7 +136,8 @@ <h3 class="panel-title">&nbsp;</h3>
         </div>
       </div>
       <div class="panel-body">
-        <div class="{{prefix}}documents-container" ct-spinner ct-spinner-id="list">
+        <div class="text-center" ng-if="listPaginated.length === 0">There are no documents</div>
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html]: can't you use a ng-switch in the parent?

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/key-dates.html [hunk 5]
```diff
@@ -85,6 +86,7 @@ <h3 class="panel-title pull-left" ng-bind="(filterParams.date && filterParams.da
         </div>
       </div>
       <ul class="{{prefix}}list-key-dates" ng-show="filterParams.date != 'dateRange'" ct-spinner ct-spinner-id="dateList">
+        <div class="text-center" ng-if="dateListFiltered.length === 0">There are no dates</div>
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/key-dates.html]: you are putting a `<div>` into a `<ul>`

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/tasks.html [hunk 6]
```diff
@@ -88,6 +88,7 @@ <h3 class="panel-title">&nbsp;</h3>
           </div>
         </div>
         <ul class="{{prefix}}list-task">
+          <div class="text-center" ng-if="listFiltered.length === 0">There are no tasks</div>
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/tasks.html]: this is invalid markup

File: uk.co.compucorp.civicrm.tasksassignments/views/modal/taskMigrate.html [hunk 7]
```diff
@@ -40,6 +41,7 @@ <h4 class="modal-title">Migrate Tasks</h4>
             <div class="col-xs-12 col-sm-8">
                 <div class="input-group">
                     <ui-select
+                            theme="civihr-ui-select"
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/views/modal/taskMigrate.html]: the indentation in this file is wrong

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html [hunk 8]
```diff
@@ -134,7 +136,8 @@ <h3 class="panel-title">&nbsp;</h3>
...
+        <div ng-show="listPaginated.length > 0" class="{{prefix}}documents-container" ct-spinner ct-spinner-id="list">
             <table class="table table-striped {{prefix}}table-documents">
               <thead>
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html]: the indentation is wrong here

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss (hunk 1 above)
Comment (deb1990) [reply uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss]: The original style it is overriding is https://github.com/civicrm/civihr/blob/PCHR-3061-ta-admin-styles/org.civicrm.bootstrapcivihr/scss/components/civihr-ui-select/_overrides.scss#L57.

But I have changed to code to target the only selector which was not having proper border color.
```
  .ui-select-match-item {
    border-color: $brand-primary !important;
  }
```

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss (hunk 2 above)
Comment (deb1990) [reply uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/modules/_civihr-ui-select.scss]: I already mentioned it in the documentation, please check.

> it overrides the css from sites/all/modules/civicrm/css/civicrm.css

```
.crm-container div.select2-container {
    font-size: 11px;
}
```

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_lists.scss (hunk 3 above)
Comment (deb1990) [reply uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_lists.scss]: Fixed by adding `ct-no-result` class.

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/tasks.html (hunk 6 above)
Comment (deb1990) [reply uk.co.compucorp.civicrm.tasksassignments/views/dashboard/tasks.html]: Fixed.

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/key-dates.html (hunk 5 above)
Comment (deb1990) [reply uk.co.compucorp.civicrm.tasksassignments/views/dashboard/key-dates.html]: Fixed.

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html (hunk 4 above)
Comment (deb1990) [reply uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html]: I have changed the structure of the HTML, now the ng-if and the ng-show are not under same parent, so cant use ng-switch. The structure was changed because `{{prefix}}list-no-result` has a margin top, and `panel-body` also has padding top, so keeping `{{prefix}}list-no-result` inside creates double spacing.
